backend/packages/Upgrade/src/api/controllers/validators/MarkExperimentValidator.ts

```diff
@@ -1,22 +1,29 @@
-import { IsNotEmpty, IsDefined } from 'class-validator';
+import { IsNotEmpty, IsDefined, IsString, IsOptional, IsUUID, IsEnum } from 'class-validator';
 import { MARKED_DECISION_POINT_STATUS } from 'upgrade_types';
 
 export class MarkExperimentValidator {
+  @IsString()
   public partitionId: string | undefined;
 
   @IsNotEmpty()
   @IsDefined()
+  @IsString()
   public experimentPoint: string;
 
   @IsNotEmpty()
   @IsDefined()
+  @IsString()
   public userId: string;
 
-  @IsNotEmpty()
-  @IsDefined()
+  @IsOptional()
+  @IsString()
   public condition: string | null;
```

Comment:
For optional change the type to condition?: string;

Reply:
Code updated